File: tests/fixtures/cpp-pilot/host/src/invoice.cpp
#include "cpppilot/invoice.hpp"

#include "cpppilot/render.hpp"
#include "invoice_internal.hpp"

namespace cpppilot {
namespace detail {

// decision: keep label policy private to this translation unit.
const char* invoice_label(int id)
{
    return id == 42 ? "INV-42" : "INV-OTHER";
}

}  // namespace detail

std::string status_name(InvoiceStatus status)
{
    switch (status) {
    case InvoiceStatus::pending:
        return "pending";
    case InvoiceStatus::paid:
        return "paid";
    }
}

std::string render_invoice(const Invoice& invoice)
{
    return join_parts(
        "invoice:", detail::invoice_label(invoice.id()), ":",
        status_name(invoice.status()), ":", CPPPILOT_MODE
    );
}

std::string render_invoice(int id, InvoiceStatus status)
{
    return render_invoice(Invoice{id, status});
}

}  // namespace cpppilot

File: tests/fixtures/kotlin-pilot/host/src/main/kotlin/kotlinpilot/Invoice.kt
package kotlinpilot

enum class InvoiceStatus {
    PENDING,
    PAID,
}

data class Invoice(val id: Int, val status: InvoiceStatus)

private fun invoiceLabel(id: Int): String =
    if (id == 42) "INV-42" else "INV-OTHER"

fun renderInvoice(invoice: Invoice): String =
    "invoice:${invoiceLabel(invoice.id)}:${invoice.status.name.lowercase()}:kotlin"

fun renderInvoice(id: Int, status: InvoiceStatus): String =
    renderInvoice(Invoice(id, status))
